azurerm/resource_arm_batch_pool.go

```diff
@@ -404,6 +451,12 @@ func resourceArmBatchPoolUpdate(d *schema.ResourceData, meta interface{}) error
 
 		parameters.PoolProperties.StartTask = startTask
 	}
+	certificates := d.Get("certificate").([]interface{})
+	certificateReferences, err := azure.ExpandBatchPoolCertificateReferences(certificates)
+	if err != nil {
+		return fmt.Errorf("Error updating Batch pool %q (Resource Group %q): %+v", poolName, resourceGroup, err)
```

Comment:
(as above) this is an error expanding the `certificate` block as such we probably want this to be:

```suggestion
		return fmt.Errorf("Error expanding `certificate`: %+v", err)
```